File: packages/elixir-sdk/test/adapters/workspaces_test.exs
defmodule TreeDxSdk.WorkspacesAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Workspaces.create(client, "repo/a", %{})
    TreeDxSdk.Workspaces.get(client, "ws/a")
    TreeDxSdk.Workspaces.close(client, "ws/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :post and &1.path == "/api/v1/workspaces/ws%2Fa/close")
           )
  end
end

File: packages/ts-sdk/test/adapters/federation/workspaces.test.ts
import { describe, expect, it } from 'vitest';
import { WorkspacesAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('WorkspacesAdapter', () => {
  it('constructs workspace endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new WorkspacesAdapter({ transport });
    await adapter.create('repo', {});
    await adapter.get('ws');
    await adapter.close('ws');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/workspaces',
      'GET /api/v1/workspaces/ws',
      'POST /api/v1/workspaces/ws/close'
    ]);
  });
});
